File: src/main/java/org/appsugar/archetypes/domain/Role.java
package org.appsugar.archetypes.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.*;

/**
 * @author shenliuyang
 * @version 1.0.0
 * @package org.appsugar.archetypes.domain
 * @className Role
 * @date 2021-03-29  22:29
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = Role.TABLE_NAME)
@DynamicUpdate
public class Role {
    public static final String TABLE_NAME = "APPSUGAR_ROLE";
    @Id
    @org.springframework.data.annotation.Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
    private Long id;
    private String name;
}

File: src/main/kotlin/org/appsugar/archetypes/entity/Role.kt
package org.appsugar.archetypes.entity

import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import org.hibernate.annotations.DynamicUpdate
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity


@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
@Entity
@DynamicUpdate
class Role(
        override var id: Long = 0,
        override var createdAt: Date = Date(),
        override var updatedAt: Date = Date(),
        var name: String = "",
        @get:Column(columnDefinition = "TEXT")
        var permissions: MutableList<String> = mutableListOf()
) : IdEntityable()